applicationinsights/channel/AsynchronousSender.py

```diff
@@ -110,6 +110,12 @@ def _run(self):
             local_send_time = local_send_interval
         while True:
             while True:
+                # Don't even bother processing if buffer is less than 100 items (the default batch size)
+                # This prevents us from wasting time sending tiny batches of data
+                # fixme magic number
+                if local_queue._queue.qsize() < 100:
```

Comment:
This is here because in my testing, performance degraded severely when small batches were allowed to be sent. Even if there was a lot of data coming in, the batches would still be small. There's probably a root cause there I could look into more. It would be easier though to do a timer. Something like "If batch size < 100 and last_send_time < 30000ms ago: skip"